editor/gui/editor_file_dialog.cpp

```diff
@@ -2326,6 +2326,10 @@ EditorFileDialog::EditorFileDialog() {
 
 	makedialog = memnew(ConfirmationDialog);
 	makedialog->set_title(TTR("Create Folder"));
+#if !defined(WIN32) && !defined(X11_ENABLED)
+	// On WinOS and Linux X11: Windows with the 'on top' can't become transient.
+	makedialog->set_flag(FLAG_ALWAYS_ON_TOP, true);
```

Comment:
This is definitly not a correct fix, being `exclusive` child (which it already is) should be enough to keep it on top of parent. 

The issue seems to be relevant only for embedded windows, so it's something wrong with subwindow handling in the `Viewport` code, and should be fixed there oinstead of adding random unrelated flags to the `Window`.

Reply:
Probably in `Viewport::_sub_window_update_order()`, seems like it's not checking for exclusive children at all.

Reply:
The following solution works with 2 `sub_windows`. 
Over the next few days, I will first check the effects on the editor GUI.

```cpp
	if (!gui.sub_windows[gui.sub_windows.size() - 1].window->get_flag(Window::FLAG_ALWAYS_ON_TOP)) {
		int index = gui.sub_windows.size() - 1;

		while (index > 0 && gui.sub_windows[index - 1].window->get_flag(Window::FLAG_ALWAYS_ON_TOP)) {
			--index;
		}

		//NEW CODE
		//+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
		Window *exclusive_child = gui.sub_windows[index].window->get_exclusive_child();
		if (exclusive_child != nullptr) {
			while (index > 0 && gui.sub_windows[index - 1].window == exclusive_child) {
				--index;
			}
		}
		//-----------------------------------------------------------------------------------

		if (index != (gui.sub_windows.size() - 1)) {
			SubWindow sw = gui.sub_windows[gui.sub_windows.size() - 1];
			gui.sub_windows.remove_at(gui.sub_windows.size() - 1);
			gui.sub_windows.insert(index, sw);
		}
	}
```